### PR TITLE
feat(utils): additional set operations

### LINKED ISSUE #8: add set intersection
We need a `set.intersection()` method.

It will help simplify this code https://github.com/tlsnotary/tlsn/blob/ce80e75461fe425bce5f66cec7b8e2770c5247bc/tlsn/tlsn-core/src/transcript.rs#L118-L127

### PR DESCRIPTION
**Pull request by sinui0**
Closes #8 

This PR adds the set intersection operation and additional subset implementations. I've also removed the superset operation which is redundant with subset.

Exhaustively testing these operations is difficult so I've added more fuzz targets, and included a proof for set-set intersection.

### CODE REVIEW

File: utils/src/range/subset.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            return false;
+        }
+
+        for other in &other.ranges {
+            if self.start >= other.end {
+                // self is rightward of other, proceed to next other
+                continue;
+            } else {
+                return self.is_subset(other);
+            }
```
Comment (th4s) [inline utils/src/range/subset.rs:25]: There is also the case of `self.end <= other.start`.

File: utils/src/range/subset.rs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        }
+
+        false
```
Comment (th4s) [inline utils/src/range/subset.rs:28]: Shouldn't this whole block rather be:
```rust
for other in &other.ranges {
  if self.is_subset(other) {
    return true;
  }
}
false
```

**Review by th4s — CHANGES_REQUESTED** [review]
Good work :+1:

~~I think there is an issue with the subset implementation.~~
 
EDIT: I forgot about the invariants of `RangeSet`.

**Review by th4s — APPROVED** [review]

**Comment by themighty1** [comment]
ACK

**Pull request merged**